features/step_definitions/service_startup.rb

```diff
@@ -132,6 +132,7 @@ module {
     if ENV["MERLIN_ENCRYPTED"] == "TRUE" then
       configfile += "\tencrypted = 1\n"
       configfile += "\tpublickey = #{ENV["MERLIN_PUBKEY"]}\n"
+      configfile += "\taccept_runcmd = 1\n"
```

Comment:
Rather than specifying this setting here, one could do it directly in the test i.e in the "I start naemon with merlin nodes connected". That would probably allow us to increase the test coverage a little and have tests for all permutations of encryption on/off and accep_runcmd on/off, without too much effort.

Reply:
Pushed a commit, is that sort of in line with what you meant?

Reply:
Roughly what I meant, if that makes sense.

I don't think it should be necessary to make in changes in the service_startup.rb file at all, it should be possible to define it in the test case directly as below:

```
@encrypted
	Scenario: The peer sends a request to "ipc" to execute a RUNCMD without accept_runcmd set
		The peer should recieve a response back.

		And I start naemon with merlin nodes connected
			| type   | name        | port |
			| peer   | my_peer     | 4123 |
		And I wait for 1 second

		Then my_peer is connected to merlin
		When my_peer sends event RUNCMD_PACKET
			| sd      | 29		 							       |
			| content | command.name=test_command;command.command_line=echo OK;runcmd=test_command |
		And I wait for 4 second

		Then file merlin.log matches RUNCMD: accept_runcmd config setting not set for this node

@encrypted
	Scenario: The peer sends a request to "ipc" to execute a RUNCMD with accept_runcmd set
		The peer should recieve a response back.

		And I start naemon with merlin nodes connected
			| type   | name        | port | accept_runcmd |
			| peer   | my_peer     | 4123 | 1                   |
		And I wait for 1 second

		Then my_peer is connected to merlin
		When my_peer sends event RUNCMD_PACKET
			| sd      | 29		 							       |
			| content | command.name=test_command;command.command_line=echo OK;runcmd=test_command |
		And I wait for 4 second
	Then my_peer received event RUNCMD_PACKET contains
			| sd      | 29	      |
			| content | outstd=OK |
			
	Scenario: The peer sends a request to "ipc" to execute a RUNCMD without accept_runcmd set, without encryption
		IPC should reject request

		And I start naemon with merlin nodes connected
			| type   | name        | port |
			| peer   | my_peer     | 4123 |
		And I wait for 1 second

		Then my_peer is connected to merlin
		When my_peer sends event RUNCMD_PACKET
			| sd      | 29		 							       |
			| content | command.name=test_command;command.command_line=echo OK;runcmd=test_command |
		And I wait for 4 second

		Then file merlin.log matches RUNCMD: some err message

	Scenario: The peer sends a request to "ipc" to execute a RUNCMD with accept_runcmd set, without encryption
		IPC should reject request

		And I start naemon with merlin nodes connected
			| type   | name        | port | accept_runcmd |
			| peer   | my_peer     | 4123 | 1                    |
		And I wait for 1 second

		Then my_peer is connected to merlin
		When my_peer sends event RUNCMD_PACKET
			| sd      | 29		 							       |
			| content | command.name=test_command;command.command_line=echo OK;runcmd=test_command |
		And I wait for 4 second

		Then file merlin.log matches RUNCMD: some err message
```

Reply:
Ahhh, I see now how the fields are parsed.